lib/index.coffee

```diff
@@ -19,7 +19,7 @@ getUnusedTreeLeafs = (tree, result = []) ->
 				getUnusedTreeLeafs(child, result)
 	return result
 
-getImagesToRemove = (tree, reclaimSpace) ->
+getImagesToRemove = (tree, reclaimSpace, gcMetrics) ->
```

Comment:
The metrics object doesn't need to be passed as parameter, and we should avoid it until we have a reason to do so. We've made an explicit choice to have it available as a singleton wherever `@balena/node-metrics-gatherer` has been `require()`'d. So, simply require at the top of the file as `metrics` and let's use it that way.

Reply:
There's one `@balena/node-metrics-gatherer` `MetricsGatherer` object per node process, that is. Any code which is running in a node process can add metrics onto it, and any http handler to export which has been set up will export all metrics from that node process.

Reply:
(a bit of background)

Reply:
That was my first try when I started this (would make my life much easier), but it didn't work because was trying to create two different endpoints in the same port. I'll try again, if it doesn't work I'll ping you in FD

Reply:
I just retried requiring using the latest version and worked like a charm. I guess it didn't work before because I tried to use the same node-metrics-gatherer version as the builder uses (which is very old) to avoid having to modify it as well. Having said that, I refactored the code to simplify it